src/commands/sites/sites-create.js

```diff
@@ -0,0 +1,187 @@
+
+  let { accountSlug } = options
+  if (!accountSlug) {
+    const { accountSlug: accountSlugInput } = await inquirer.prompt([
+      {
+        type: 'list',
+        name: 'accountSlug',
+        message: 'Team:',
+        choices: accounts.map((account) => ({
+          value: account.slug,
+          name: account.name,
+        })),
+      },
+    ])
+    accountSlug = accountSlugInput
+  }
+
+  const { name: nameFlag } = options
+  let user
+  let site
+
+  // Allow the user to reenter site name if selected one isn't available
+  const inputSiteName = async (name) => {
+    if (!user) user = await api.getCurrentUser()
+
+    if (!name) {
+      let { slug } = user
+      let suffix = ''
+
+      // If the user doesn't have a slug, we'll compute one. Because `full_name` is not guaranteed to be unique, we
+      // append a short randomly-generated ID to reduce the likelihood of a conflict.
+      if (!slug) {
+        slug = slugify(user.full_name || user.email)
+        suffix = `-${uuidv4().slice(0, SITE_NAME_SUGGESTION_SUFFIX_LENGTH)}`
+      }
+
+      const suggestions = [
+        `super-cool-site-by-${slug}${suffix}`,
+        `the-awesome-${slug}-site${suffix}`,
+        `${slug}-makes-great-sites${suffix}`,
+        `netlify-thinks-${slug}-is-great${suffix}`,
+        `the-great-${slug}-site${suffix}`,
+        `isnt-${slug}-awesome${suffix}`,
+      ]
+      const siteSuggestion = sample(suggestions)
+
+      console.log(
+        `Choose a unique site name (e.g. ${siteSuggestion}.netlify.app) or leave it blank for a random name. You can update the site name later.`,
+      )
+      const { name: nameInput } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'name',
+          message: 'Site name (optional):',
+          filter: (val) => (val === '' ? undefined : val),
+          validate: (input) => /^[a-zA-Z\d-]+$/.test(input) || 'Only alphanumeric characters and hyphens are allowed',
+        },
+      ])
+      name = nameInput
+    }
+
+    const body = {}
+    if (typeof name === 'string') {
+      body.name = name.trim()
+    }
+    try {
+      site = await api.createSiteInTeam({
+        accountSlug,
+        body,
+      })
+    } catch (error_) {
+      if (error_.status === 422) {
+        warn(`${name}.netlify.app already exists. Please try a different slug.`)
+        await inputSiteName()
+      } else {
+        error(`createSiteInTeam error: ${error_.status}: ${error_.message}`)
+      }
+    }
+  }
+  await inputSiteName(nameFlag)
+
+  log()
+  log(chalk.greenBright.bold.underline(`Site Created`))
+  log()
+
+  const siteUrl = site.ssl_url || site.url
+  log(
+    prettyjson.render({
+      'Admin URL': site.admin_url,
+      URL: siteUrl,
+      'Site ID': site.id,
+    }),
+  )
+
+  track('sites_created', {
+    siteId: site.id,
+    adminUrl: site.admin_url,
+    siteUrl,
+  })
+
+  if (options.withCi) {
+    log('Configuring CI')
+    const repoData = await getRepoData()
```

Comment:
```suggestion
    const repoData = await getRepoData({})
```

Unrelated bug introduced in https://github.com/netlify/cli/pull/2827

Reply:
I would solve it like this in `src/utils/get-repo-data.js`:

```
/**
 *
 * @param {object} config
 * @param {string} [config.remoteName]
 * @returns
 */
const getRepoData = async function ({ remoteName } = {}) {
```